### PR TITLE
Popups close themselves

### PR DESCRIPTION
**Pull request by nickspoons**
Recreated PR #310 which was closed prematurely.

Please feel free to close this if this functionality is not desired.

### CODE REVIEW

File: autoload/vista/popup.vim [hunk 1]
```diff
@@ -8,7 +8,10 @@ let s:popup_delay = get(g:, 'vista_floating_delay', 100)
 
 function! s:ClosePopup() abort
   if exists('s:popup_winid')
-    call popup_hide(s:popup_winid)
+    call popup_close(s:popup_winid)
+    try
+      unlet s:popup_winid
```
Comment (liuchengxu) [inline autoload/vista/popup.vim:13]: This try-catch should be unnecssary.

I can merge this if it does work for you, other people might also benefit from it :(.

Comment (nickspoons) [reply autoload/vista/popup.vim]: Yeah you're right. I added it after hitting an error there at one stage, but that was while debugging and re-sourcing the script a lot, I suspect I just got the plugin into an invalid state, I can't reproduce it now.

**Comment by nickspoons** [comment]
Feedback applied, thanks

**Comment by liuchengxu** [comment]
Thanks!

**Pull request merged**